Feature flags, public access and plans

nokome opened the pull request:
This PR started as an implementation of per-user feature flags and a "waiting room" page (so we can gradually increase the number of users who have access to full functionality). However, after considering how to deal with public projects (anon users should be able to list and download files, snapshots etc) and with invites (invited users shouldn't go on a waiting list), it seemed better to instead focus on implementing account tiers and quotas (we can initially set the maximum number of projects that an account can create to zero).

@alex-ketch Could you please finish off a first iteration of the `account/plan.html` template. See `class AccountTier` for the metrics that can be included and `class AccountQuotas` for how they are calculated. I have made a very rough start on it just to sketch out what data to send through to the template context. I envisaged that users would be able to switch plans on the same page and see a listing of subscriptions (although I am yet to reinstate the Stripe integration).

alex-ketch commented:
@nokome I've pushed up some style/layout changes. Working on the template I think there are a couple of fields which would be useful to be added to the tier model:
- price
- name
- public project limit

Don't remember if included or, but a creation date or similar field would allow us to keep around legacy tiers for select accounts.

nokome commented:
Thanks @alex-ketch. I have replaced `projects_total` with `projects_public` and added `name` and `job_runtime_month` fields. I have not added a `price` because that ill come when the Stripe integration is reinstated.
